lib/grpc/client/adapters/mint/stream_response_process.ex

```diff
@@ -0,0 +1,199 @@
+defmodule GRPC.Client.Adapters.Mint.StreamResponseProcess do
+  @moduledoc false
+  # This module represents the process responsible for consuming the
+  # incoming messages from a connection. For each request, there will be
+  # a process responsible for consuming its messages. At the end of a stream
+  # this process will automatically be killed.
+
+  # TODO: Refactor the GenServer.call/3 occurrences on this module to produce
+  # telemetry errors in case of failures
+
+  @typep accepted_types :: :data | :trailers | :headers | :error
+  @typep data_types :: binary() | Mint.Types.headers() | Mint.Types.error()
+
+  @accepted_types [:data, :trailers, :headers, :error]
+  @header_types [:headers, :trailers]
+
+  use GenServer
```

Comment:
Aren't we missing a `@behaviour GenServer` here?

I'd expect `handle_continue` & friends to warn on missing `@impl true` below

Reply:
yeah! that's strange indeed hahaha. I'll annotate the `@impl` on the bellow  tho